### PR TITLE
14164 Fixed a few BC/CCC/ULC things

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#14164

*Description of changes:*
- deleted CCC workaround
- added new types in redirectThisFiling()
- added entity types to allowed corrections
- added "isCcc" to allowable actions mixin
- added resource objects for BC/CC/ULC
- added "isCcc" store getter
- updated unit tests
- app version = 5.7.19

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -146,12 +146,12 @@ export default {
...
         CorpTypeCd.BENEFIT_COMPANY,
+        CorpTypeCd.BC_CCC,
         CorpTypeCd.BC_COMPANY,
         CorpTypeCd.BC_ULC_COMPANY,
-        CorpTypeCd.BC_CCC,
         CorpTypeCd.COOP,
-        CorpTypeCd.SOLE_PROP,
-        CorpTypeCd.PARTNERSHIP
+        CorpTypeCd.PARTNERSHIP,
+        CorpTypeCd.SOLE_PROP
```
Comment (severinbeauvais) [inline src/App.vue:154]: Sorted.

File: src/App.vue [hunk 2]
```diff
@@ -676,11 +676,7 @@ export default {
       }
 
       // verify that NR type matches entity type from application
-      if (nr.legalType === 'CCC' && this.getEntityType === CorpTypeCd.BC_CCC) {
-        // TEMPORARY FIX:
-        // at the moment, Namex API is passing CCC instead of CC
-        // remove this when no longer needed
```
Comment (severinbeauvais) [inline src/App.vue:682]: No longer needed. Removed.

File: src/mixins/allowable-actions-mixin.ts [hunk 3]
```diff
@@ -8,17 +8,18 @@ import { AllowableActions, CorpTypeCd, Routes } from '@/enums'
...
   @Getter isHistorical!: boolean
+  @Getter isPartnership!: boolean
   @Getter isRoleStaff!: boolean
   @Getter isSoleProp!: boolean
-  @Getter isPartnership!: boolean
-  @Getter isFirm!: boolean
-  @Getter isCoop!: boolean
-  @Getter hasBlockerExceptStaffApproval!: boolean
-  @Getter isGoodStanding!: boolean
+  @Getter isUlc!: boolean
```
Comment (severinbeauvais) [inline src/mixins/allowable-actions-mixin.ts:22]: Sorted.

File: src/mixins/resource-lookup-mixin.ts [hunk 4]
```diff
@@ -27,6 +27,7 @@ export default class ResourceLookupMixin extends Vue {
 
     /**
      * Returns the current entity's full display name.
+     * Used by Correction component.
```
Comment (severinbeauvais) [inline src/mixins/resource-lookup-mixin.ts:30]: We really should name the resource objects in a way that denotes what they're used for. Future.

File: src/resources/business-config.ts [hunk 5]
```diff
@@ -284,5 +285,221 @@ export const ConfigJson = [
...
+      entityTitle: 'Company',
+      subTitle: 'an incorporated company',
+      act: 'Business Corporations',
+      modalTitle: 'Voluntary Dissolution',
+      confirmButtonText: 'Continue with Voluntary Dissolution',
+      dissolutionType: 'voluntarily dissolve'
+    },
+    todoList: {
+      title: 'Voluntary Dissolution'
+    }
```
Comment (severinbeauvais) [inline src/resources/business-config.ts:503]: These 3 new objects are copies of the BEN object except for entity type, display name, and number of directors (CCC only). I create ticket 14315 for UI team to review this and let us know if it needs updating.

File: src/resources/business-config.ts [hunk 6]
```diff
@@ -13,6 +13,7 @@ export class Business {
   flows: Array<Flow>
 }
 
+// FUTURE: this object needs an interface or type
```
Comment (ChrisGabel23) [inline src/resources/business-config.ts:16]: Is FUTURE different from a TODO?

**Review by ChrisGabel23 — APPROVED** [review]

File: src/resources/business-config.ts (hunk 6 above)
Comment (severinbeauvais) [reply src/resources/business-config.ts:16]: For me, yes. (We have not previously discussed this as a group.)

TODO = a comment to myself to do this before I commit my changes (in scope of current ticket)

FUTURE = a comment to me/others that this should be looked at some time (not in scope of current ticket); sometimes I'll create a ticket for such items, sometimes not

**Review by lewischenstudio — APPROVED** [review]
LGTM

**Pull request merged**